### PR TITLE
Only show on production if user is super admin

### PR DESCRIPTION
**Pull request by stuartshields**
@jazzsequence Based on our earlier discussions, here is some code that _should_ only enable it on production. I have looked at the code a lot this morning as I had a weird bug locally that was unrelated to I might have missed something.

However the idea is that:
If the current environment is production and the user isn't a super_admin then it won't display the field.

### CODE REVIEW

**Comment by stuartshields** [comment]
@jazzsequence So I ended up removing `$creds_defined` check because it was giving false positives. Actually added in a check to bail if the creds aren't defined. We shouldn't be waiting longer and do it sooner.

File: inc/namespace.php [hunk 1]
```diff
@@ -60,31 +65,47 @@ function is_development_environment() : bool {
...
 }
 
+/**
+ * Enable settings on production.
+ *
+ * @return bool
+ */
+function enable_auth(): bool {
+	if (
+		! is_super_admin() && defined( 'HM_ENV_TYPE' ) && HM_ENV_TYPE === 'production' ) {
```
Comment (jazzsequence) [inline inc/namespace.php]: spacing here is weird. either put all the conditions on the same line
```php
if ( ! is_super_admin() && defined( 'HM_ENV_TYPE' ) && HM_ENV_TYPE === 'production' )
```
or break it up on a separate line
```php
if (
    ! is_super_admin() && defined( 'HM_ENV_TYPE' ) && HM_ENV_TYPE === 'production' 
)
```
or break each condition into its own line
```php
if (
    ! is_super_admin() && 
    defined( 'HM_ENV_TYPE' ) && 
    HM_ENV_TYPE === 'production' 
)
```

```suggestion
		! is_super_admin() && defined( 'HM_ENV_TYPE' ) && HM_ENV_TYPE === 'production' 
	) {
```

File: inc/namespace.php [hunk 2]
```diff
@@ -45,6 +44,12 @@ function is_development_environment() : bool {
 	 */
 	$other_dev = apply_filters( 'hmauth_filter_dev_env', false );
 
+	// Bail early if the credentials are missing.
+	$creds_defined = defined( 'HM_BASIC_AUTH_USER' ) && defined( 'HM_BASIC_AUTH_PW' );
+	if ( ! $creds_defined ) {
```
Comment (jazzsequence) [inline inc/namespace.php:50]: 👍

**Review by jazzsequence — CHANGES_REQUESTED** [review]
One issue with spacing that's fairly minor. However, I want to be clear about what would cause the setting to be available on a production environment and what would enable basic auth in that case...

If
 * we're on production
 * AND we have credentials defined for production
 * OR we're filtering to add credentials on production
then
 * the setting is visible to _super admins_
 * AND only _super admins_ can enable or disable basic authentication
 * AND basic authentication extends only to the current site

If
 * we're on production
 * AND no credentials are defined in the config for production (either globally or for the environment)
 * AND we're not using the filter
then
 * the setting does not appear in production environments for _any users_
 * AND Basic Auth is not able to be activated on non-production environments
 * AND Basic Auth still works as normal on dev/staging environments

If that matches with your expectations and what you're seeing in testing, I think this is probably good to go after fixing the spacing issue. I was going to push back harder about allowing Basic Auth on production environments because I feel like that is best handled through the filter, since it's something that would be pretty specific to the environment/site, but I think there are enough protections and checks in place here that _in most cases_ it's still disabled on production, and just allows for it to be enabled on production environments more easily.

**Comment by jazzsequence** [comment]
Looks like there's some linting issues. I will clean this up and push out a release today.

**Review by jazzsequence — APPROVED** [review]

**Pull request merged**